integration/spark2/src/main/scala/org/apache/spark/sql/hive/ICarbonSessionCatalog.scala

```diff
@@ -0,0 +1,67 @@
+/*
+* Licensed to the Apache Software Foundation (ASF) under one or more
+* contributor license agreements.  See the NOTICE file distributed with
+* this work for additional information regarding copyright ownership.
+* The ASF licenses this file to You under the Apache License, Version 2.0
+* (the "License"); you may not use this file except in compliance with
+* the License.  You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+package org.apache.spark.sql.hive
+
+import org.apache.hadoop.fs.Path
+import org.apache.spark.sql.{CarbonEnv, SparkSession}
+import org.apache.spark.sql.catalyst.TableIdentifier
+import org.apache.spark.sql.catalyst.catalog.{CatalogStorageFormat, CatalogTablePartition}
+import org.apache.spark.sql.catalyst.expressions.Expression
+
+import org.apache.carbondata.common.annotations.{InterfaceAudience, InterfaceStability}
+
+/**
+ * This interface defines those common api used by carbon for spark-2.1 and spark-2.2 integration,
+ * but are not defined in SessionCatalog or HiveSessionCatalog to give contract to the
+ * Concrete implementation classes.
+ * For example CarbonSessionCatalog defined in 2.1 and 2.2.
+ *
+ */
+@InterfaceAudience.Internal
+@InterfaceStability.Stable
+trait ICarbonSessionCatalog {
```

Comment:
I think the best approach to verify whether it is really abstracted is to verify it with spark-2.3, can you check once by adding profile for spark-2.3 and report in this PR

Reply:
Fixed renames ICarbonSessionCatalog  to CarbonSessionCatalog 
and CarbonSessionCatalog  to ICarbonHiveSessionCatalog